sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -303,7 +298,8 @@ final class DataFrameWriter private[sql](df: DataFrame) {
       conn.close()
     }
 
-    JdbcUtils.saveTable(df, url, table, props)
+    val propsForSave = JDBCRelation.getEffectiveProperties(connectionProperties, this.extraOptions)
```

Comment:
Sorry, I may still not understand: why can't you use `props` here? you're copying the `Properties` twice, it seems.